### PR TITLE
W:12689961: Add ServerVariables to Components in  Async 2.4

### PR DESCRIPTION
**Pull request by damianpedra**

### CODE REVIEW

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/WebApiReferenceResolutionStage.scala [hunk 1]
```diff
@@ -45,6 +45,13 @@ class WebApiReferenceResolutionStage(keepEditingInfo: Boolean = false)
               copy.withId(sourceServer.id).withName(sourceServer.name.value())
             case _ => domain
           }
+        case sourceServerVariable: Parameter =>
```
Comment (looseale) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/WebApiReferenceResolutionStage.scala]: Add the check for the annotation DeclaredServerVariable

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async20ServerVariableParser.scala [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import amf.core.internal.parser.domain.{Annotations, ScalarNode, SearchScope}
+import amf.core.internal.utils.IdCounter
+import amf.core.internal.validation.CoreValidations
+import amf.shapes.client.scala.model.domain.Example
+import amf.shapes.internal.domain.metamodel.common.ExamplesField
+import amf.shapes.internal.spec.common.parser.YMapEntryLike
+import amf.shapes.internal.validation.definitions.ShapeParserSideValidations.ExamplesMustBeASeq
+import org.yaml.model.{YMap, YMapEntry, YNode, YSequence, YType}
+
+case class Async20ServerVariableParser(entry: YMapEntryLike, parent: String)(implicit override val ctx: AsyncWebApiContext)
```
Comment (looseale) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async20ServerVariableParser.scala]: The name change in the class is ok, but revert the name change in the file

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/WebApiReferenceResolutionStage.scala (hunk 1 above)
Comment (damianpedra) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/stages/WebApiReferenceResolutionStage.scala]: done

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async20ServerVariableParser.scala (hunk 2 above)
Comment (damianpedra) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/async/parser/domain/Async20ServerVariableParser.scala]: done

**Review by looseale — APPROVED** [review]

**Pull request merged**